authorize/check_response.go

```diff
@@ -25,15 +26,15 @@ import (
 
 func (a *Authorize) okResponse(reply *evaluator.Result) *envoy_service_auth_v3.CheckResponse {
 	var requestHeaders []*envoy_config_core_v3.HeaderValueOption
-	for k, v := range reply.Headers {
-		requestHeaders = append(requestHeaders, mkHeader(k, v, false))
+	for k, vs := range reply.Headers {
+		requestHeaders = append(requestHeaders, mkHeader(k, strings.Join(vs, ","), false))
```

Comment:
Should we also sort the header key values to ensure determinism?

Reply:
we already sort the headers after this loop.